### PR TITLE
Fix/16961 - Update existing events

### PR DESCRIPTION
**Pull request by nikola-milicevic**
Part of [#16961](https://github.com/wordpress-mobile/WordPress-iOS/issues/16961)

This PR updates Reader tracking events, so now they match proposal found in the spreadsheet shared on p9xfpQ-1eg-p2.

## Notes
- All reader tracking events (excluding `reader_reader_tag_followed` / `reader_reader_tag_unfollowed`) now contain: `blog_id`, `feed_id` and `follow` properties.
- `source` property will be added in a separate PR

## Review notes
- It is best to review this PR by going from commit to commit chronologically, since every commit addresses needed changes for a particular tracking event (including adding needed properties for a event, removing double event, fixing mistyped event, adding new event)

## To test
1. Trigger every event found in the events table found [here](https://github.com/wordpress-mobile/WordPress-Android/pull/14283) (with the generosity of @ParaskP7). Trigger can be found in the Trigger column.
2. Make sure tracking event log in the console includes: `blog_id`, `feed_id` and `follow` properties.

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by momo-ozawa** [comment]
@nikola-milicevic This PR's milestone is set to 18.4 but the target branch is `epic/16961-update-reader-tracking-properties`. Did you mean to target `develop`?

File: WordPress/Classes/ViewRelated/Reader/ReaderSaveForLater+Analytics.swift
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Reader/ReaderSaveForLater+Analytics.swift]: Typo 👀 
```suggestion
                                         "follow": post.isFollowing]
```

**Review by momo-ozawa — COMMENTED** [review]
@nikola-milicevic 

~`reader_site_followed` and `reader_site_unfollowed` don't have the `feed_id` included in the event log.~

~According to [this chart](https://github.com/wordpress-mobile/WordPress-Android/pull/14283#issue-835664355), it looks like `feed_id` should be included.~

Nvm -- read your comment!
> All reader tracking events (excluding reader_site_followed / reader_site_unfollowed and reader_reader_tag_followed / reader_reader_tag_unfollowed) now contain: blog_id, feed_id and follow properties.

File: WordPress/Classes/ViewRelated/Reader/ReaderSaveForLater+Analytics.swift
Comment (nikola-milicevic) [reply WordPress/Classes/ViewRelated/Reader/ReaderSaveForLater+Analytics.swift]: Oops 😬 I will definitely go for a enum key in a separate PR, but thanks for catching this anyway!

**Comment by nikola-milicevic** [comment]
> @nikola-milicevic This PR's milestone is set to 18.4 but the target branch is `epic/16961-update-reader-tracking-properties`. Did you mean to target `develop`?

@momo-ozawa  Well not sure what's correct here. Here I want to target `epic` definitely, since I will have (at least) two more PR, and then once all is approved / merged into `epic`, I'll merge `epic` into `develop`. How to make the bots happy if I omit the milestone?

File: WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m
Comment (leandroalonso) [inline WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m:1493]: Important info: we almost never rename events. The reason is that there's no "merge" in the Analytics tool, so if we do it we end up with two different events and that can possibly affect a lot of calculations, funnels, etc.

In Android the name of the event is `reader_blog_previewed` but AFAIK it has always been like that. It's a shame that both events doesn't have the same name but we should keep `reader_blog_preview`.

Comment (leandroalonso) [inline WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m:1493]: Important info: we almost never rename events. The reason is that there's no "merge" in the Analytics tool, so if we do it we end up with two different events and that can possibly affect a lot of calculations, funnels, etc.

In Android the name of the event is `reader_blog_previewed` but AFAIK it has always been like that. It's a shame that both events don't have the same name but we should keep `reader_blog_preview`.

File: WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m
Comment (leandroalonso) [inline WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m]: Same as above.

File: Podfile.lock
Comment (leandroalonso) [inline Podfile.lock:14]: Are these `Podfile.lock` changes intentional?

File: WordPress/Classes/Services/ReaderPostService.h
Comment (leandroalonso) [inline WordPress/Classes/Services/ReaderPostService.h:110]: Maybe we can rename `details` to `isFromArticleDetails`? Reading the method signature is clear what that means, but in the implementation inside the method `details` is not clear.

**Review by leandroalonso — CHANGES_REQUESTED** [review]
@nikola-milicevic about:

> All reader tracking events (excluding reader_site_followed / reader_site_unfollowed and reader_reader_tag_followed / reader_reader_tag_unfollowed) 

`reader_site_followed` and `reader_site_unfollowed` should have a `feedId` (based on the table from Android's PR).

A few other notes:

* `reader_article_rendered` doesn't seem to exist on iOS. I wonder if it's an event that it's triggered when the article finishes rendering through the webview? Not sure if we should implement that, wdyt?
* `reader_blog_previewed` is duplicated, eg.:

```
2021-09-24 10:39:36:525 WordPress[3506:54946] 🔵 Tracked: reader_blog_previewed <blog_id: 184797733, follow: 0, is_jetpack: 0, post_id: 9569, URL: https://sawidji.com>
2021-09-24 10:39:36:966 WordPress[3506:54946] 🔵 Tracked: reader_blog_previewed <blog_id: 184797733, feed_id: 116888526, follow: 0, site: Sawidji Gallery, source: reader, subscription_count: 8>
```

This is unrelated to this task but it's clearly a bug, we can tackle that on another PR.

**Comment by leandroalonso** [comment]
> Well not sure what's correct here. Here I want to target epic definitely, since I will have (at least) two more PR, and then once all is approved / merged into epic, I'll merge epic into develop. How to make the bots happy if I omit the milestone?

You can add the `Pending` or `Someday` milestones. :)

File: WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m
Comment (nikola-milicevic) [reply WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m:1493]: Hmm well then we have `reader_blog_preview` and `reader_blog_previewed` events, since there are two enums defined in the app, called from different places, which are identically named ("-previewed"), but have different values (["preview"](https://github.com/wordpress-mobile/WordPress-iOS/blob/c464da08602a3eab58d3a2987eff72821648147c/WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m#L1492) and ["previewed"](https://github.com/wordpress-mobile/WordPress-iOS/blob/cfdfab82158289015985bd47536a896dc4f40f1a/WordPress/Classes/Utility/Analytics/WPAnalyticsEvent.swift#L289)), so it looked clearly as a typo to me.
Should I revert the change I made?

File: WordPress/Classes/Services/ReaderPostService.h
Comment (nikola-milicevic) [reply WordPress/Classes/Services/ReaderPostService.h:110]: Actually I would avoid doing this since I think it is not in line with Objective-C naming conventions. Few examples: [here](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingMethods.html#//apple_ref/doc/uid/20001282-BCIGIJJF), [here](https://google.github.io/styleguide/objcguide.html) and [here](https://stackoverflow.com/questions/15719854/naming-convention-for-local-bool-variables-in-objective-c-and-cocoa). I agree that `details` is not the most clear name, though I would not use `isFromArticleDetails` which is more Swifty. What do you think?

File: Podfile.lock
Comment (nikola-milicevic) [reply Podfile.lock:14]: I just updated the pods, nothing more. Should I revert this?

Comment (leandroalonso) [reply Podfile.lock:14]: I wouldn't update them if it's not strictly needed for this PR. I know it's just a minor version but we avoid doing so.

Plus, AppCenter is normally handled by the release team.

File: WordPress/Classes/Services/ReaderPostService.h
Comment (leandroalonso) [reply WordPress/Classes/Services/ReaderPostService.h:110]: Legibility > conventions IMHO. But this method is not big, so I don't think it's a big problem, you can leave it as it is.

**Comment by nikola-milicevic** [comment]
@leandroalonso 

> `reader_site_followed` and `reader_site_unfollowed` should have a feedId

On it.

> `reader_article_rendered` doesn't seem to exist on iOS.

I'll add it.

> `reader_blog_previewed` is duplicated

Good catch. But how can I reproduce it? Since it didn't happen during my testing.

File: WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m
Comment (leandroalonso) [reply WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m:1493]: Ughhhh, you're right. In that case, the ship has sailed, I agree with you, let's leave as it is.

File: WordPress/Classes/Services/ReaderPostService.h
Comment (nikola-milicevic) [reply WordPress/Classes/Services/ReaderPostService.h:110]: > Legibility > conventions

I agree, but someone coming from Obj-C might not 😊 I'm leaving it as is for now.

File: WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m
Comment (nikola-milicevic) [reply WordPress/Classes/Utility/Analytics/WPAnalyticsTrackerAutomatticTracks.m]: I'm reverting this back to `reader_tag_preview`.

**Comment by leandroalonso** [comment]
> Good catch. But how can I reproduce it? Since it didn't happen during my testing.

This was probably fixed in 6fbe41da8eda5ec7a827b8bf5620b2fe02ef98ba, I'll test again once it's ready for another review. :)

**Comment by nikola-milicevic** [comment]
@momo-ozawa @leandroalonso I updated the code, thanks for the comments!

**Review by leandroalonso — COMMENTED** [review]
@nikola-milicevic all good. The only pending item is to move the new event to the `WPAnalyticsEvent.swift`.

**Comment by nikola-milicevic** [comment]
@leandroalonso Thanks! I update the code. Should be good to go now.

**Review by leandroalonso — APPROVED** [review]

**Pull request merged**
